### PR TITLE
Improved logging for Signatures and POSes

### PR DESCRIPTION
**Pull request by lynxart**
Added Signature and POS logging functionality on request of my Eve University corpmates. Signature create/edit/bulk import now results in log entries per signature, and I tried to differentiate between real edits (with meaningful changes/complete scans) and simple "click edit and submit" action. Also includes distance from root system in the logging.
For POSes it does a similar thing, this required some changes in the URLs - now passing mapsystem ID along instead of system (logging requires the active map to be known).

### CODE REVIEW

**Comment by nyrocron** [comment]
Can this be disabled in case one doesn't need this functionality and wants to save the server some work?

**Comment by lynxart** [comment]
What would you like to see disabled specifically? I'm a bit surprised that this would have any significant effect on performance... 
If you tell me what things are most taxing I will see if I can make them conditional.

File: evewspace/Map/models.py [hunk 1]
```diff
@@ -761,6 +771,80 @@ def update(self):
...
+        COL_SIG_GROUP = 2
+        COL_SIG_SCAN_GROUP = 1
+        COL_SIG_STRENGTH = 4
+        COL_DISTANCE = 5
+        info = row[COL_SIG_TYPE]
+        sig_type = None
+        updated = None
+        action = "None"
+
+        if wascreated is True:
```
Comment (nyrocron) [inline evewspace/Map/models.py]: should be "if wascreated"

File: evewspace/Map/models.py [hunk 2]
```diff
@@ -597,6 +597,16 @@ def has_siblings(self):
             return False
         return parent_sys.childsystems.count() > 1
 
+    def distance_from_root(self):
```
Comment (nyrocron) [inline evewspace/Map/models.py]: calculating this for each edit could be a performance issue with multiple concurrent users editing signatures

File: evewspace/Map/views.py [hunk 3]
```diff
@@ -654,26 +552,39 @@ def bulk_sig_import(request, map_id, ms_id):
...
                 sig_id = utils.convert_signature_id(row[COL_SIG])
-                sig = Signature.objects.get_or_create(
-                    sigid=sig_id, system=map_system.system)[0]
-                sig = _update_sig_from_tsv(sig, row)
+                # Query signatures in system to see if it exists
+                sig, sigcreated = Signature.objects.get_or_create(
+                    sigid=sig_id, system=map_system.system)
+
+                # Update sig with copied TSV data
+                sig, update_type = sig.update_from_tsv(request.user, sigcreated, row, map_system)
```
Comment (nyrocron) [inline evewspace/Map/views.py]: line > 80 characters

File: evewspace/POS/templatetags/poslist.py [hunk 4]
```diff
@@ -13,17 +13,20 @@
...
+	system = get_object_or_404(System, pk=mapsystem.system.pk)
+	poses = POS.objects.filter(system=system)
+	return {'poses': poses, 'mapsystem': mapsystem, 'request': context['request']}
 
 
 @register.inclusion_tag('posdetails_small.html', takes_context=True)
-def posdetails(context, system, pos):
-    return {'pos' : pos, 'system': system, 'perms': context['perms']}
+def posdetails(context, mapsystem, pos):
+	return {'pos' : pos, 'mapsystem': mapsystem, 'perms': context['perms']}
```
Comment (nyrocron) [inline evewspace/POS/templatetags/poslist.py]: indent should be 4 spaces

**Comment by nyrocron** [comment]
Commented on some of the issues in the code right now. There are several other PEP8 violations and little things that could be improved. If i have time tomorrow i'll try to fix some of them.

**Comment by lynxart** [comment]
As I'm new to this kind of collaboration on github, another question: should I fix the issues you highlighted and push them to my develop branch? Or should I let you do the fixes?

**Comment by Zumochi** [comment]
@lynxart it's easiest for you to do the fixes yourself if you can. Otherwise @nyrocron could make a PR to your fork.
If you add commits to a branch you have a PR for they'll automatically show up in the PR.

**Comment by lynxart** [comment]
OK, thanks. I will take care of the suggested edits.

File: evewspace/Map/models.py (hunk 2 above)
Comment (lynxart) [inline evewspace/Map/models.py]: Distance calculation is now only used on signature edits if the MAP_ADVANCED_LOGGING setting is active (it's a new setting in General Settings).

**Comment by nyrocron** [comment]
sounds good

**Comment by marbindrakon** [comment]
I went ahead and merged this for now. I think logging is ultimately something that needs to be heavily redone.

**Pull request merged**